src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -109,4 +140,67 @@ public function transChoice($id, $number, array $parameters = array(), $domain =
         return (bool) preg_match_all(static::$regexSprintfParams, $string)
             && !(bool) preg_match_all(static::$regexClassicParams, $string);
     }
+
+    /**
+     * Tries to translate the provided message using the legacy system
+     *
+     * @param string $message
+     * @param array $parameters
+     * @param string $domain
+     * @param string|null $locale
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     */
+    private function translateUsingLegacySystem($message, array $parameters, $domain, $locale = null)
+    {
+        $domainParts = explode('.', $domain);
+        if (count($domainParts) < 2) {
+            throw new InvalidArgumentException(sprintf('Invalid domain: "%s"', $domain));
+        }
+
+        $moduleName = strtolower($domainParts[1]);
+        $sourceFile = (!empty($domainParts[2])) ? strtolower($domainParts[2]) : $moduleName;
+
+        // translate using the legacy system WITHOUT fallback to the new system (to avoid infinite loop)
+        return (new LegacyTranslator())->translate($moduleName, $message, $sourceFile, $parameters, false, $locale, false);
+    }
+
+    /**
+     * Indicates if we should try and translate the provided wording using the legacy system.
+     *
+     * @param string $message Message to translate
+     * @param string $domain Translation domain
+     * @param string $translated Message after first translation attempt
+     *
+     * @return bool
+     */
+    private function shouldFallbackToLegacyModuleTranslation($message, $domain, $translated)
+    {
+        return
+            $message === $translated
+            && 'Modules.' === substr($domain, 0, 8)
```

Comment:
it's not enough and it break the front end (try to login)

Reply:
It doesn't anymore